src/main/java/seedu/address/logic/parser/PayCommandParser.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.PayCommand.MESSAGE_USAGE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOUR;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.PayCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new PayCommand object.
+ */
+public class PayCommandParser implements Parser<PayCommand> {
+
+    private static final Logger logger = Logger.getLogger(PayCommandParser.class.getName());
+
+    /**
+     * Parses input arguments and creates a new PayCommand object.
+     */
+    public PayCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        logger.log(Level.INFO, "Parsing PayCommand with arguments: " + args);
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_HOUR);
+
+        Index index;
+        double hour;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            logger.log(Level.WARNING, "Failed to parse index. Invalid command format.", pe);
```

Comment:
The logger statements used throughout the code base was 
```
logger.info("----------------[USER COMMAND][" + commandText + "]");
logger.warning("Error reading from jsonFile file " + filePath + ": " + e);
```

I think it is better to be consistent throughout and use that format instead of `logger.log`. It is also good to include information of the preamble in the log message.

Reply:
Thanks for the reminder, Ive standardised the logger